dm/master/coordinator/coordinator.go

```diff
@@ -492,9 +498,9 @@ func (c *Coordinator) restartMysqlTask(w *Worker, cfg *config.MysqlConfig) bool
 	if w.State() == WorkerClosed {
 		ectx, cancel := context.WithTimeout(c.etcdCli.Ctx(), etcdTimeout)
 		defer cancel()
-		resp, err := c.etcdCli.Get(ectx, common.WorkerRegisterKeyAdapter.Encode(w.Address(), w.Name()))
+		resp, err := c.etcdCli.Get(ectx, common.WorkerKeepAliveKeyAdapter.Encode(w.Address(), w.Name()))
 		if err != nil {
-			if resp.Count > 0 {
+			if resp != nil && resp.Count > 0 {
```

Comment:
`if err != nil`, can `resp != nil` be true?

Reply:
emmmm. I think it should be
`if err != nil ` -> `if err == nil`